client/src/pages/dashboard/conversations_list.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useCallback, useEffect } from "react"
+import { useLocalStorage } from "usehooks-ts"
+import { Box, Flex, Text } from "theme-ui"
+
+import { useAppDispatch, useAppSelector } from "../../hooks"
+import { RootState } from "../../store"
+import { ConversationSummary, populateConversationsSummary } from "../../reducers/conversations_summary"
+import { useHistory } from "react-router-dom"
+
+const ConversationsList = ({selectedConversationId}: {selectedConversationId: string | null}) => {
+  const dispatch = useAppDispatch()
+  const hist = useHistory()
+
+  const {data} = useAppSelector((state: RootState) => state.conversations_summary)
+  const conversations = data || []
+
+  const [showAllFIPConversations, setShowAllFIPConversations] = useLocalStorage("showAll", true)
+  const [showOpenFIPConversations, setShowOpenFIPConversations] = useLocalStorage("showOpen", false)
+  const [showNonFIPConversations, setShowNonFIPConversations] = useLocalStorage("showNonFIP", false)
+  const [showArchivedConversations, setShowArchivedConversations] = useLocalStorage(
+    "showArchived",
+    false,
+  )
+  const [showHiddenConversations, setShowHiddenConversations] = useLocalStorage("showHidden", false)
```

Comment:
These selections are mutually exclusive so we could simplify the code a lot here